src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/AppContext.java

```diff
@@ -38,6 +38,10 @@ public AppContext() {
     this.clusterDetailsEventProcessor = null;
   }
 
+  public AppContext(AppContext other) {
+    this.clusterDetailsEventProcessor = new ClusterDetailsEventProcessor(other.clusterDetailsEventProcessor);
```

Comment:
should we also can the default constructor this() here in case something else needs to be initialized in AppContext ?

Reply:
This is a copy constructor. So ideally it should copy everything but after rebasing I saw, you added a new member which is a cache. I am sure we don't want to make a copy of it. 
So, I will call `this()` as you sugested

Reply:
Thinking more about it, I will explicitly initialize the other. Calling other() initializes the other with default which we may not intend.